src/main/sensors/battery.c

```diff
@@ -227,15 +236,36 @@ static void batteryUpdateVoltageState(void)
     switch (voltageState) {
         case BATTERY_OK:
             if (voltageMeter.filtered <= (batteryWarningVoltage - batteryConfig()->vbathysteresis)) {
-                voltageState = BATTERY_WARNING;
+                if (start_t == 0) {
+                    start_t = millis();
+                }
+                end_t = millis();
+                total_t = (end_t - start_t)/1000;
```

Comment:
Why are you introducing global variables for `end_t` and `total_t`?

Reply:
You mean lines 72-74 right? So I should initialize these variables in 240-243 instead?

Reply:
I am not sure they are even needed.